aws/sdk/examples/config/src/bin/enable-config.rs

```diff
@@ -0,0 +1,187 @@
+///   If not supplied, uses the value of the **AWS_REGION** environment variable.
+///   If the environment variable is not set, defaults to **us-west-2**.
+/// * `[-v]` - Whether to display information.
+/// Need: s3 key prefix AND kms key
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    tracing_subscriber::fmt::init();
+    let Opt {
+        region,
+        bucket,
+        iam_arn,
+        kms_arn,
+        name,
+        prefix,
+        sns_arn,
+        type_,
+        verbose,
+    } = Opt::from_args();
+
+    let region = region::ChainProvider::first_try(region.map(Region::new))
+        .or_default_provider()
+        .or_else(Region::new("us-west-2"));
+
+    println!();
+
+    if verbose {
+        println!("Config client version:          {}", PKG_VERSION);
+        println!(
+            "Region:                         {}",
+            region.region().unwrap().as_ref()
+        );
+        println!("Resource type:                  {}", type_);
+        println!("Config (delivery channel) name: {}", name);
+        println!("Bucket:                         {}", bucket);
+        println!("Prefix:                         {}", prefix);
+        println!("SNS ARN:                        {}", sns_arn);
+        println!("IAM ARN:                        {}", iam_arn);
+        println!("KMS ARN:                        {}", kms_arn);
+        println!();
+    }
+
+    let conf = Config::builder().region(region).build();
+    let client = Client::from_conf(conf);
+
+    // If we already have a configuration recorder in the Region, we cannot create another.
+    let resp = client.describe_configuration_recorders().send().await?;
+
+    let recorders = resp.configuration_recorders.unwrap_or_default();
+
+    let num_recorders = recorders.len();
+
+    if num_recorders != 0 {
+        println!("You already have a configuration recorder in this region");
+        println!("Use delete-configuration-recorder to delete it before you call this again.");
+
+        for recorder in recorders {
+            println!("Recorder: {}", recorder.name.as_deref().unwrap_or_default());
+        }
+
+        process::exit(1);
+    }
+
+    // If we already have a delivery channel in the Region, we cannot create another.
+    let resp = client.describe_delivery_channels().send().await?;
+
+    let channels = resp.delivery_channels.unwrap_or_default();
+
+    let num_channels = channels.len();
+
+    if num_channels != 0 {
+        println!("You already have a delivery channel in this region");
+        println!("Use delete-delivery-channel to delete it before you call this again.");
+
+        for channel in channels {
+            println!("  Channel: {}", channel.name.as_deref().unwrap_or_default());
+        }
+
+        process::exit(1);
+    }
+
+    let resource_types: Vec<ResourceType> = vec![ResourceType::Topic];
+
+    let rec_group = RecordingGroup::builder()
+        .set_resource_types(Some(resource_types))
+        .build();
```

Comment:
you don't need to go via `Vec` here, you can do:
```rust
RecordingGroup::builder().resource_types(ResourceType::Topic).build()
```

Reply:
I'll update the code.